nx/lib/nx.ex

```diff
@@ -7229,6 +7231,72 @@ defmodule Nx do
     )
   end
 
+  @doc """
+  Calculates the Singular Value Decomposition of 2-D tensors.
+
+  If `compute_uv = true`, returns `{u, s, vt}`. Otherwise, returns `s`.
+  `s` is an array containing the singular values of the input tensor.
+  `u` and `vt` are orthonormal matrices where `input_tensor = u.S.vt`,
+  and `S` is the diagonal matrix which elements are the elements of `s`.
+
+  The elemens of `s` are sorted from highest to lowest.
+
+  ## Options
+    * `compute_uv`: `boolean`. Defaults to `false`
+      Whether or not to compute `u` and `vt`. If `false`, the return value is `s`
+
+  ## Examples
+      iex> Nx.svd(Nx.tensor([[1, 0, 0], [0, 1, 0], [0, 0, -1]]), compute_uv: true)
+      {
+        Nx.tensor([
+          [1.0, 0.0, 0.0],
+          [0.0, 1.0, 0.0],
+          [0.0, 0, 1.0]
+        ]),
+        Nx.tensor([1.0, 1.0, 1.0]),
+        Nx.tensor([
+          [1.0, 0.0, 0.0],
+          [0.0, 1.0, 0.0],
+          [0.0, 0, -1.0]
+        ])
+      }
```

Comment:
Instead of checking against the tuple, you can match on it and compare each element individually:

```
      iex> {u, s, vt} = Nx.svd(Nx.tensor([[1, 0, 0], [0, 1, 0], [0, 0, -1]]), compute_uv: true)
      iex> u
      #Nx.Tensor<...>
      iex> s
      #Nx.Tensor<...>
     iex> vt
      #Nx.Tensor<...>
```

Note you can't have empty lines between them for ExUnit to consider them the same doctest. :)

Reply:
Oh, I thought you had to number them like `iex(1)`, `iex(2)`, etc. Nice to know :)